pom.xml

```diff
@@ -1,55 +1,193 @@
+    <junit.version>5.5.1</junit.version>
+
+    <!-- Maven Plugins -->
+    <maven-clean-plugin.version>3.1.0</maven-clean-plugin.version>
+    <maven-enforcer-plugin.version>3.0.0-M3</maven-enforcer-plugin.version>
+    <maven-resources-plugin.version>3.2.0</maven-resources-plugin.version>
+    <maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
+    <maven-surefire-plugin.version>3.0.0-M5</maven-surefire-plugin.version>
+    <maven-dependency-plugin.version>3.1.2</maven-dependency-plugin.version>
+    <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
+    <maven-source-plugin.version>3.2.1</maven-source-plugin.version>
+    <maven-javadoc-plugin.version>3.2.0</maven-javadoc-plugin.version>
+    <maven-install-plugin.version>3.0.0-M1</maven-install-plugin.version>
+    <maven-gpg-plugin.version>1.6</maven-gpg-plugin.version>
+    <maven-deploy-plugin.version>3.0.0-M1</maven-deploy-plugin.version>
+    <maven-site-plugin.version>3.9.1</maven-site-plugin.version>
+
+    <!-- Other Plugins -->
+    <flatten-maven-plugin.version>1.2.5</flatten-maven-plugin.version>
+    <enforcer-extra-rules.version>1.3</enforcer-extra-rules.version>
+    <versions-maven-plugin.version>2.8.1</versions-maven-plugin.version>
+    <jacoco-maven-plugin.version>0.8.6</jacoco-maven-plugin.version>
+    <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
+
+    <!-- Enabled for Releases -->
+    <maven.javadoc.skip>true</maven.javadoc.skip>
+    <maven.source.skip>true</maven.source.skip>
+
+    <jacoco.exec.datafile>${project.build.directory}/coverage-reports/jacoco-ut.exec</jacoco.exec.datafile>
   </properties>
 
-  <dependencies>
-    <dependency>
-      <groupId>org.projectlombok</groupId>
-      <artifactId>lombok</artifactId>
-      <version>${lombok.version}</version>
-      <scope>provided</scope>
-    </dependency>
-    <dependency>
-      <groupId>com.github.docker-java</groupId>
-      <artifactId>docker-java</artifactId>
-      <version>${docker-java.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.github.docker-java</groupId>
-      <artifactId>docker-java-api</artifactId>
-      <version>${docker-java.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>com.github.docker-java</groupId>
-      <artifactId>docker-java-core</artifactId>
-      <version>${docker-java.version}</version>
-    </dependency>
-  </dependencies>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.projectlombok</groupId>
+        <artifactId>lombok</artifactId>
+        <version>${lombok.version}</version>
+        <scope>provided</scope>
+      </dependency>
+
+      <!-- BOMs -->
+      <dependency>
+        <groupId>com.fasterxml.jackson</groupId>
+        <artifactId>jackson-bom</artifactId>
+        <version>${jackson.version}</version>
+        <scope>import</scope>
+        <type>pom</type>
+      </dependency>
+      <dependency>
```

Comment:
This one should not be in the parent pom or it might kill the junit4 module.
I've faced unpredictable issues when both junit 4 and junit5 end up in the classpath.

Reply:
> I've faced unpredictable issues when both junit 4 and junit5 end up in the classpath.

But they never do. Junit5 has different coordinates, because this is declared under `dependencyManagement` the only way a project will import any of the dependencies declared in the `junit-bom` is if it explicitly declares a dependency therein.

In the  Junit 4 module we do:
```
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>${junit4.version}</version>
    </dependency>
```
which will _only_ import Junit 4.

In the Junit 5 module we do:
```
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-api</artifactId>
    </dependency>
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-engine</artifactId>
      <scope>runtime</scope>
    </dependency>
```
whcih will _only_ import Junit 5 (at the version defined in the `junit-bom` declared in the parent pom.

The reason I declared the Junit 5 in the parent is because both `core` and `consul` should use Junit 5 for the UTs.

For the Junit4 module if indeed there are incompatibilities then we will have no choice but to write any UTs (if needed) with Junit 4.

Reply:
You are right. I didn't notice that it was inside the `<dependencyManagement>` section.
Now I understand what you want to achieve. JUnit5 by default for all tests and JUnit4 in the JUni4 module.